### ISSUE TITLE
`uv tool` should support man pages

### ISSUE THREAD

**Issue by zanieb** [astral-sh/uv#4731]
pipx supports installation of man pages for documentation. We should do the same in the `uv tool` commands.

**Comment by harkabeeparolus** [comment]
Agreed. This is the biggest feature that's still keeping me on pipx right now.

**Comment by eth3lbert** [comment]
> Agreed. This is the biggest feature that's still keeping me on pipx right now.

Before this feature is fully implemented, you can use the following workaround: `uv tool run --from package man package`.

e.g., `uv tool run --from pycowsay man pycowsay`.

**Comment by mattiasb** [comment]
> Before this feature is fully implemented, you can use the following workaround: 
> `uv tool run --from package man package`.
> 
> e.g., `uv tool run --from pycowsay man pycowsay`.

Or you could add this to your `~/.bashrc`:
```bash
# Add uv tool man-files to MANPATH
MANPATH="$(find "${XDG_DATA_HOME:-${HOME}/.local/share}/uv/tools" \
                -maxdepth 3 -type d -name man -printf '%p:')"
export MANPATH
```

This makes me able to read man pages in the regular manner for the tools I've installed that ships man pages.

**NOTE**: I'm neither an expert on `man` nor `uv` so there might be corner cases that I haven't understood.

**EDIT**: Made the script more robust against repetitions.

**Comment by harkabeeparolus** [comment]
>     find "${XDG_DATA_HOME:-${HOME}/.local/share}/uv/tools"                     \
>          -maxdepth 3                                                           \
>          -type d                                                               \
>          -name man                                                             \
>          -printf '%p:'

I mean... That's... Wow... It's a thing. 😬

My _$MANPATH_ would be really ugly using this strategy with some of the tools I have installed, e.g. _basedpyright_. If you install it with **pipx**, it correctly adds zero (0) manpages. Using **uv** with the strategy above, it would add 3 useless MANPATH directories from the NodeJS dependency, for a total of 87 (!) unrelated manpages. 😮

I'd rather just keep using pipx for any CLI tools with manpages for now. 😉

And it would be super nice if uv did this correctly. 😊

**Comment by mattiasb** [comment]
> I mean... That's... Wow... It's a thing. 😬

It's a *workaround* not a solution. Please treat it as such.
 
> My _$MANPATH_ would be really […] for a total of 87 (!) unrelated manpages. 😮

You missed the `-maxdepth 3` flag.

**Comment by harkabeeparolus** [comment]
Yeah, you're right. 😳 I apologize for overreacting! 😅

**Comment by callegar** [comment]
Even for a workaround, why not try symlinking man pages into a single hierarchy to be placed in manpath? Ideally, this could be `.local/share/man`, but if this is considered dangerous, even `.local/share/uv/man` or `.local/share/uv/tools_man` would help, letting one add a single fixed entry to the manpath.

**Comment by harkabeeparolus** [comment]
Yes, ideally it should be `~/.local/share/man`, since the `manpath` utilty on Linux, FreeBSD, and macOS automatically picks up this directory if `~/.local/bin` is on your `$PATH`. See for example:

* https://man.freebsd.org/cgi/man.cgi?query=manpath
* https://man7.org/linux/man-pages/man5/manpath.5.html

**Comment by stdedos** [comment]
> > Before this feature is fully implemented, you can use the following workaround:
> > `uv tool run --from package man package`.
> > e.g., `uv tool run --from pycowsay man pycowsay`.
> 
> Or you could add this to your `~/.bashrc`:
> 
> # Add uv tool man-files to MANPATH
> MANPATH="$(find "${XDG_DATA_HOME:-${HOME}/.local/share}/uv/tools" \
>                 -maxdepth 3 -type d -name man -printf '%p:')"
> export MANPATH
> This makes me able to read man pages in the regular manner for the tools I've installed that ships man pages.
> 
> **NOTE**: I'm neither an expert on `man` nor `uv` so there might be corner cases that I haven't understood.
> 
> **EDIT**: Made the script more robust against repetitions.

Even better:

```sh
# Add uv tool man-files to MANPATH
MANPATH="$(find "${UV_TOOL_DIR:-${XDG_DATA_HOME:-${HOME}/.local/share}/uv/tools}" -maxdepth 3 -type d -name man -printf '%p:')${MANPATH:+$MANPATH}"
export MANPATH="${MANPATH%:}"
```

preserve `MANPATH` if already set; trim trailing `:`
